frontend/src/types/Task.ts

```diff
@@ -1,9 +1,19 @@
 export default interface Task {
+
   id?: number;
   name: string;
   description?: string;
   progress: number;
   weight: number;
-  completedAt: Date | null;
-  children: Task[] | null;
+  completedAt?: Date | null;
```

Comment:
since these are made optional with the `?`, the `| null` is not necessary (unless we want this data to be the data or undefined or null, but probably not)